### PR TITLE
Update Trainer code example

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

This PR updates the code example used in the Trainer docs. Previously, it shows how to overwrite the Trainer to do multi-label classification. However, this is not required anymore, as users can now pass the `problem_type` argument to the model's configuration (to use the appropriate loss function).

Instead, I show how to overwrite the Trainer to use a weighted loss, useful when you have an imbalanced dataset.

### CODE REVIEW

File: docs/source/main_classes/trainer.mdx [hunk 1]
```diff
@@ -47,22 +47,22 @@ when you use it on other models. When using it on your own model, make sure:
...
     def compute_loss(self, model, inputs, return_outputs=False):
         labels = inputs.get("labels")
+        # forward pass
         outputs = model(**inputs)
         logits = outputs.get("logits")
-        loss_fct = nn.BCEWithLogitsLoss()
-        loss = loss_fct(
-            logits.view(-1, self.model.config.num_labels), labels.float().view(-1, self.model.config.num_labels)
-        )
+        # compute custom loss
```
Comment (sgugger) [inline docs/source/main_classes/trainer.mdx:63]: Maybe add a mention that you have three labels here, so that's why three weights?

**Review by sgugger — APPROVED** [review]
Thanks a lot for rewriting this example!
